Implement rpc handler to record worker heartbeat.

rkannan82 opened the pull request:
## What changed?
Implement the rpc handler that creates/updates the worker chasm component with the new heartbeat.

## Why?

## How did you test it?
- [ ] built
- [ ] run locally and tested manually
- [ ] covered by existing tests
- [x] added new unit test(s)
- [ ] added new functional test(s)

## Potential risks
None. Feature under developement.

rkannan82 committed:
Implement rpc handler to record worker heartbeat.

bergundy commented on chasm/lib/worker/handler_test.go:
```diff
@@ -0,0 +1,105 @@
+	testWorkerInstanceKey = "test-worker-1"
+)
+
+func TestRecordHeartbeatHandler(t *testing.T) {
```
Make this a functional test, all of these mocks don't prove that this API actually works.

rkannan82 committed:
Pass full request proto to recordHeartbeat.

rkannan82 committed:
Reduce boilerplate in calling UpdateWithNewEntity

rkannan82 committed:
Remove dynamic config.

We can add this back when we make the frontend handler changes.
The frontend handler will use this flag when forwarding the heartbeat to history service.

rkannan82 committed:
Delete handler_test as it is not adding any value

rkannan82 reviewed (commented):
Please take another look.

rkannan82 committed:
Add functional test

rkannan82 committed:
Add verification for functional test

bergundy commented on tests/worker_heartbeat_test.go:
```diff
@@ -0,0 +1,162 @@
+			},
+		),
+		func(w *chasmworker.Worker, _ chasm.Context, _ any) (*chasmworker.Worker, error) {
+			return w, nil
```
Avoid the data race:

```suggestion
			return w.Clone(), nil
```

rkannan82 committed:
Fix broken code after rebase - align with parent's 2-state model

rkannan82 committed:
Fix build and test errors - correct CHASM API usage

rkannan82 committed:
Fix frontend files - add worker client wiring

rkannan82 committed:
Add EnableWorkerStateTracking flag and CHASM worker heartbeat call

rkannan82 committed:
debug panic with counter and defensive code

rkannan82 committed:
UpdateWithNew not implmented
